packages/frontend/src/hooks/useIsSmartContractWallet.ts

```diff
@@ -0,0 +1,20 @@
+import { useEffect, useState } from "react";
+import { useWeb3Context } from "src/contexts/Web3Context";
+
+export default function useIsSmartContractWallet() {
+  const { provider, address } = useWeb3Context();
+  const [isSmartContractWallet, setIsSmartContractWallet] = useState(false);
+
+  useEffect(() => {
+    const checkAddress = async () => {
+      if (!provider || !address) return;
+
+      const code = await provider.getCode(address.address);
+      setIsSmartContractWallet(!code || code !== '0x');
```

Comment:
is there a case that `code` will be falsy here? (i.e. an empty string). if not, i think we can simply this for readability

```suggestion
      setIsSmartContractWallet(code !== '0x');
```